pkg/storage/tpr/storage_interface.go

```diff
@@ -135,50 +149,60 @@ func (t *store) Create(
 	return nil
 }
 
-// Delete removes the specified key and returns the value that existed at that spot.
-// If key didn't exist, it will return NotFound storage error.
+// Delete fetches the resource at key, removes its finalizer, updates it, and returns the
+// resource before its finalizer was removed.
 //
-// In this implementation, Delete will not write the deleted object back to out
+// If key didn't exist, it will return NotFound storage error.
 func (t *store) Delete(
```

Comment:
That makes sense. I didn't intuitively understand that REST vs abstraction over it was the delineation. Now that you say it, it makes sense. Cool.